src/shared/codelens/codeLensUtils.ts

```diff
@@ -100,6 +120,7 @@ function makeAddCodeSamDebugCodeLens(params: MakeAddDebugConfigCodeLensParams):
                 resourceName: params.handlerName,
                 rootUri: params.rootUri,
                 runtimeFamily: params.runtimeFamily,
+                runtime: params.runtime,
```

Comment:
CodeLenses should be codified with the bare minimum details required to take action. In this scenario (to start an "add debug config" command), the definitive details are the handler (resourceName), the manifest location (rootUri), and what language we're dealing with (runtimeFamily). Runtime is an optional and variable based on the contents of a workspace.

Is there a reason the efforts to make a template match and (best-effort) resolve a runtime are required _before_ the CodeLens has been clicked?

An additional consideration is that processing required to produce a CodeLens should be minimal/as fast as possible (per VS Code guidance). VS Code queries the extension for CodeLenses every time the document is modified.

Reply:
> Is there a reason the efforts to make a template match and (best-effort) resolve a runtime are required before the CodeLens has been clicked?

The main reason is that this leaves the door open for further linkages. Since we have a list of matching templates when we're drawing codelenses, we'd be able to draw codelenses that link back to template files. Assuming this is a feature we want in the future (I definitely want it, FWIW), doing it the other way means we'd be parsing through templates in two separate places.

I'm not too concerned re: speed since everything is in-memory at this point thanks to the registry.

The main flaw here is if someone were to have a split window with the source file and template side-by-side, and the user edits the template's runtime after the codelens has been drawn in the source file. I'm not 100% sure I care to account for this case?